derive_builder/examples/validator_fn.rs

```diff
@@ -0,0 +1,37 @@
+//! This example illustrates the use of `validator` to add a pre-build validation
+//! step.
+
+#[macro_use]
+extern crate derive_builder;
+
+#[derive(Builder, Debug, PartialEq)]
+#[builder(build_fn(validator="LoremBuilder::validate"))]
```

Comment:
Fair enough :+1:. I could just not see in what situation it would be usable. If you have a free function with generics and trait bounds, then you would have to implement traits for the builder. I can't say if that is useful, I can't come up with a use case at the moment.